diff --git a/twrp.cpp b/twrp.cpp
index c9fdc57..a8bb40a 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -115,10 +115,6 @@ static void Decrypt_Page(bool SkipDecryption, bool datamedia) {
 static void process_fastbootd_mode() {
 		LOGINFO("starting fastboot\n");
 
-		if (android::base::GetBoolProperty("ro.boot.dynamic_partitions", false)) {
-			PartitionManager.Unmap_Super_Devices();
-		}
-
 #ifdef AB_OTA_UPDATER
 		DataManager::SetValue("tw_active_slot", PartitionManager.Get_Active_Slot_Display());
 #endif
