diff --git a/gui/action.cpp b/gui/action.cpp
index 42e3079..6a67377 100755
--- a/gui/action.cpp
+++ b/gui/action.cpp
@@ -203,6 +203,8 @@ GUIAction::GUIAction(xml_node <> *node):GUIObject(node)
       // These actions will be run in the caller's thread
       ADD_ACTION(reboot);
       ADD_ACTION(home);
+      ADD_ACTION(openfox);
+      ADD_ACTION(gotofastbootd);
       ADD_ACTION(key);
       ADD_ACTION(page);
       ADD_ACTION(reload);
@@ -697,6 +699,19 @@ int GUIAction::home(std::string arg __unused)
   return 0;
 }
 
+int GUIAction::openfox(std::string arg __unused)
+{
+  gui_changePage("main");
+  return 0;
+}
+
+int GUIAction::gotofastbootd(std::string arg __unused)
+{
+  property_set("ro.orangefox.fastbootd", "1");
+  gui_changePage("fastboot");
+  return 0;
+}
+
 int GUIAction::key(std::string arg)
 {
   const int key = getKeyByName(arg);
